libqtile/backend/wayland/core.py

```diff
@@ -1449,14 +1449,20 @@ def get_screen_info(self) -> list[ScreenRect]:
 
     def grab_key(self, key: config.Key | config.KeyChord) -> tuple[int, int]:
         """Configure the backend to grab the key event"""
-        keysym = xkb.keysym_from_name(key.key, case_insensitive=True)
+        if isinstance(key.key, str):
+            keysym = xkb.keysym_from_name(key.key, case_insensitive=True)
+        else:
+            keysym = self.keyboards[0]._state.key_get_one_sym(key.key)
```

Comment:
You can get the active keyboard using the seat. Use self.seat.get_keyboard() and null check it. _state is nowhere defined anyways right?

Reply:
How does that help? the `Keyboard` returned by this doesn't have the keymap (as far as I can tell) so I can't get a keysym from the code.

`self._state` is defined in this PR as part of our `Keyboard` class.

Reply:
Loop through keyboards and check if the wlr_keyboard is the right one. getting the first keyboard is wrong